### DISCUSSION TITLE
Stored Procedure works in supabase SQL shell but not with the SDK

### DISCUSSION THREAD

**Discussion by magiusdarrigo** [supabase/supabase#2896] [Questions] [answered]
Hello! I am quite new to using Supabase.

I executed this function successfully in the supabase SQL shell:
`create or replace function getNumOfDistinctMenusForRestaurant (domain_name_input text) returns int as
$BODY$
  declare 
    unique_menu_cnt int;
  begin

select count(distinct menu_uuid) into unique_menu_cnt
  from restaurants
  join locations on (restaurants.id = locations.restaurant_uuid)
  where domain_name = domain_name_input;
return unique_menu_cnt;

end;
$BODY$
language plpgsql;`

I was then able to successfully call it in the shell as such:
`SELECT * FROM getNumOfDistinctMenusForRestaurant ('mydomain.com');`

The above query gives me back the expected table.

So now I am wondering why when using the SDK:
`const { data, error } = await supabase.rpc( 'getNumOfDistinctMenusForRestaurant', {
    domain_name_input: domainName
  } )`

I get the error:
`{
  hint: 'No function matches the given name and argument types. You might need to add explicit type casts.',
  details: null,
  code: '42883',
  message: 'function public.getNumOfDistinctMenusForRestaurant(domain_name_input => text) does not exist'
}`

Thanks!

**Comment by soedirgo** [comment, marked as answer]
Postgres is case-insensitive, so you'll have to surround your function name in quotes:
```sql
create or replace function "getNumOfDistinctMenusForRestaurant" (domain_name_input text) returns int as
$BODY$
declare
unique_menu_cnt int;
begin

select count(distinct menu_uuid) into unique_menu_cnt
from restaurants
join locations on (restaurants.id = locations.restaurant_uuid)
where domain_name = domain_name_input;
return unique_menu_cnt;

end;
$BODY$
language plpgsql;
```

Let me know if it works!

**Reply by magiusdarrigo** [reply]
Wow! I learned something new, it works perfectly now. Thank you!
